### PR TITLE
feat(jans): jetty 11 integration

### PR DESCRIPTION
**Pull request by pujavs**
Code changes for jetty upgrade from 9.x to 11.0.8

Refer related discussion [951](https://github.com/JanssenProject/jans/discussions/951)

### CODE REVIEW

File: jans-auth-server/server/pom.xml [hunk 1]
```diff
@@ -373,23 +373,23 @@
...
             <groupId>net.bootsfaces</groupId>
             <artifactId>bootsfaces</artifactId>
             <version>1.4.1</version>
             <scope>compile</scope>
         </dependency>
-
+ -->
 			<!-- OmniFaces -->
-		<dependency>
+		<!-- <dependency>
```
Comment (yurem) [inline jans-auth-server/server/pom.xml]: We need to check if we not use this library in all xhtml (including custom authenticators). if we use it we need to use pure JSF

File: jans-auth-server/pom.xml [hunk 2]
```diff
@@ -17,17 +17,24 @@
...
+		<servlet.version>5.0.0</servlet.version>
+
 		<activemq.version>5.15.14</activemq.version>
 
 		<jettison.version>1.4.1</jettison.version>
 
 		<slf4j.version>1.7.25</slf4j.version>
 
-		<omnifaces.version>2.6.9</omnifaces.version>
+		<!-- <omnifaces.version>2.6.9</omnifaces.version> -->
```
Comment (yurem) [inline jans-auth-server/pom.xml]: We need to check if we not use this library in all xhtml (including custom authenticators). if we use it we need to use pure JSF

File: jans-bom/pom.xml [hunk 3]
```diff
@@ -13,19 +13,47 @@
...
+		<weld.version>4.0.3.Final</weld.version>
 		<richfaces.version>4.5.19-gluu.Final</richfaces.version>
-
-		<jsf-api.version>2.2.20</jsf-api.version>
-		<jsf-impl.version>2.3.14</jsf-impl.version>
-		<uel-api.version>2.2.1-b04</uel-api.version>
-		<uel-impl.version>2.2.1-b05</uel-impl.version>
+		
+		<!-- Jakarta -->
+		<jakarta.jakartaee.api.version>9.0.0</jakarta.jakartaee.api.version>
```
Comment (yurem) [inline jans-bom/pom.xml]: This section is over-managed. The main part here which specify compatible versions is Weld. Hence we import weld-core-parent [pom](https://mvnrepository.com/artifact/org.jboss.weld/weld-core-parent/4.0.3.Final) file. It should provides information about compatible versions. If we need to override specific libs we need to add comments to this definition to explain the reason of this override. Now we can get into similar to DLL hell situation.

File: jans-config-api/pom.xml [hunk 4]
```diff
@@ -22,12 +22,33 @@
...
+		<jakarta.ws.rs.api.version>3.0.0</jakarta.ws.rs.api.version>
+		<org.hibernate.validator.version>7.0.1.Final</org.hibernate.validator.version>
+		<org.hibernate.validator.cdi.version>7.0.1.Final</org.hibernate.validator.cdi.version>
+		<org.glassfish.jakarta.el.version>4.0.0</org.glassfish.jakarta.el.version>
+		<org.glassfish.jakarta.json.version>2.0.1</org.glassfish.jakarta.json.version>
+
 		<jackson.version>2.10.1</jackson.version>
-		<microprofile.version>3.0</microprofile.version>
-		<smallrye.version>2.2.0</smallrye.version>
+		<microprofile.config.version>3.0.1</microprofile.config.version>
```
Comment (yurem) [inline jans-config-api/pom.xml:30]: We can import microprofile [bom ](https://mvnrepository.com/artifact/org.eclipse.microprofile/microprofile) file.

File: jans-config-api/pom.xml [hunk 5]
```diff
@@ -22,12 +22,33 @@
...
+		<microprofile.health.version>4.0</microprofile.health.version>
+		<microprofile.metrics.version>4.0</microprofile.metrics.version>
+		<microprofile.rest.client.version>3.0</microprofile.rest.client.version>		
+		<smallrye.config.version>3.0.0-RC1</smallrye.config.version>
```
Comment (yurem) [inline jans-config-api/pom.xml:34]: We can import smallrye [bom ](https://mvnrepository.com/artifact/io.smallrye/smallrye-bom) file.

**Review by yurem — CHANGES_REQUESTED** [review]

**Comment by moabu** [comment]
Issue holding this PR from being merged https://github.com/JanssenProject/jans/issues/1190

**Comment by yuriyz** [comment]
Before merging this PR we need : 
1) solve ORM issue https://github.com/JanssenProject/jans/issues/1190
2) get @Milton-Ch feedback about running `AuthorizationCodeFlowHttpTest` against jetty 11 based AS.

**Comment by Milton-Ch** [comment]
image

**Comment by moabu** [comment]
ok only issue left is #1190

**Comment by moabu** [comment]
Scan on war build on 4/13/2022


| NAME                              | INSTALLED    | FIXED-IN            | VULNERABILITY       | SEVERITY |
|-----------------------------------|--------------|---------------------|---------------------|----------|
| commons-httpclient                | 3.1          | CVE-2012-5783       | Medium              |          |
| commons-httpclient                | 3.1          | CVE-2020-13956      | Medium              |          |
| docker                            | 5.0.3        | CVE-2021-21284      | Medium              |          |
| docker                            | 5.0.3        | CVE-2019-13509      | High                |          |
| docker                            | 5.0.3        | CVE-2021-21285      | Medium              |          |
| docker                            | 5.0.3        | CVE-2018-10892      | Medium              | High     |
| docker                            | 5.0.3        | CVE-2019-16884      | High                |          |
| docker                            | 5.0.3        | CVE-2019-13139      | High                |          |
| docker                            | 5.0.3        | CVE-2020-27534      | Medium              |          |
| docker                            | 5.0.3        | CVE-2019-5736       | High                |          |
| geronimo-j2ee-management_1.1_spec | 1.0.1        | CVE-2008-0732       | Low                 |          |
| geronimo-j2ee-management_1.1_spec | 1.0.1        | CVE-2011-5034       | High                |          |
| geronimo-jms_1.1_spec             | 1.1.1        | CVE-2011-5034       | High                |          |
| geronimo-jms_1.1_spec             | 1.1.1        | CVE-2008-0732       | Low                 |          |
| geronimo-jta_1.1_spec             | 1.1.1        | CVE-2011-5034       | High                |          |
| geronimo-jta_1.1_spec             | 1.1.1        | CVE-2008-0732       | Low                 |          |
| guava                             | 28.0-android | CVE-2020-8908       | Low                 | Medium   |
| guava                             | 28.0-android | GHSA-5mg8-w23w-74h3 | Low                 |          |
| jackson-databind                  | 2.12.5       | 2.12.6.1            | GHSA-57j2-w4cx-62h2 | High     |
| jackson-databind                  | 2.12.5       | CVE-2020-36518      | High                |          |
| jetty-jakarta-servlet-api         | 5.0.2        | CVE-2017-7658       | Critical            | High     |
| jetty-jakarta-servlet-api         | 5.0.2        | CVE-2017-9735       | High                |          |
| jetty-jakarta-servlet-api         | 5.0.2        | CVE-2020-27216      | High                | High     |
| jetty-jakarta-servlet-api         | 5.0.2        | CVE-2021-28169      | Medium              | High     |
| jetty-jakarta-servlet-api         | 5.0.2        | CVE-2021-34428      | Low                 | Medium   |
| jetty-jakarta-servlet-api         | 5.0.2        | CVE-2017-7656       | High                |          |
| jetty-jakarta-servlet-api         | 5.0.2        | CVE-2009-5045       | High                |          |
| jetty-jakarta-servlet-api         | 5.0.2        | CVE-2009-5046       | Medium              |          |
| jetty-jakarta-servlet-api         | 5.0.2        | CVE-2017-7657       | Critical            |          |
| jetty-jakarta-websocket-api       | 2.0.0        | CVE-2017-7656       | High                |          |
| jetty-jakarta-websocket-api       | 2.0.0        | CVE-2017-9735       | High                |          |
| jetty-jakarta-websocket-api       | 2.0.0        | CVE-2009-5046       | Medium              |          |
| jetty-jakarta-websocket-api       | 2.0.0        | CVE-2021-34428      | Low                 |          |
| jetty-jakarta-websocket-api       | 2.0.0        | CVE-2017-7658       | Critical            |          |
| jetty-jakarta-websocket-api       | 2.0.0        | CVE-2017-7657       | Critical            |          |
| jetty-jakarta-websocket-api       | 2.0.0        | CVE-2021-28169      | Medium              | Medium   |
| jetty-jakarta-websocket-api       | 2.0.0        | CVE-2009-5045       | High                | High     |
| jetty-jakarta-websocket-api       | 2.0.0        | CVE-2020-27216      | High                |          |
| jetty-openid                      | 11.0.8       | CVE-2008-3280       | Medium              | Critical |
| jetty-openid                      | 11.0.8       | CVE-2007-1651       | Medium              |          |
| jetty-openid                      | 11.0.8       | CVE-2007-1652       | High                |          |
| jetty-setuid-java                 | 1.0.4        | CVE-2021-34428      | Low                 |          |
| jetty-setuid-java                 | 1.0.4        | CVE-2020-27216      | High                | High     |
| jetty-setuid-java                 | 1.0.4        | CVE-2017-7657       | Critical            | High     |
| jetty-setuid-java                 | 1.0.4        | CVE-2009-5045       | High                | High     |
| jetty-setuid-java                 | 1.0.4        | CVE-2017-7658       | Critical            | Critical |
| jetty-setuid-java                 | 1.0.4        | CVE-2017-7656       | High                |          |
| jetty-setuid-java                 | 1.0.4        | CVE-2017-9735       | High                |          |
| jetty-setuid-java                 | 1.0.4        | CVE-2021-28169      | Medium              |          |
| jetty-setuid-java                 | 1.0.4        | CVE-2009-5046       | Medium              |          |
| kubernetes                        | 23.3.0       | CVE-2015-7561       | Low                 |          |
| kubernetes                        | 23.3.0       | CVE-2016-1905       | High                | High     |
| kubernetes                        | 23.3.0       | CVE-2020-8554       | Medium              |          |
| kubernetes                        | 23.3.0       | CVE-2021-25740      | Low                 |          |
| kubernetes                        | 23.3.0       | CVE-2016-1906       | Critical            | Low      |
| kubernetes                        | 23.3.0       | CVE-2016-7075       | High                | Medium   |
| libldap                           | 2.4.58-r0    | CVE-2015-3276       | Medium              | High     |
| libpq                             | 13.6-r0      | CVE-2017-8806       | Medium              | High     |
| netty-codec-http                  | 4.1.69.Final | 4.1.71.Final        | GHSA-wx5j-54mm-rqqq | Medium   |
| postgresql                        | 42.3.3       | CVE-2017-8806       | Medium              |          |
| xmlsec                            | 2.1.4        | 2.1.7               | GHSA-j8wc-gxx9-82hx | High     |

**Comment by manojs1978** [comment]
### issues found during retesting.

----------------------------------------------------------------------------------------------------------------------------------------------------------------------
| \#  | Module  | issues Description                                                                                                             | Status | 
| --- | ------------------------------| ---------------------------------------------------------------------------------------------------------------| ---------------|
| 1   |  UMA resource.         | 1)unable to create UMA resource getting below error,                                                    | Open            | |
|      |                                  | we need to add name and description then only                                                             |                   |
|       |                                 | we can create new resource                                                                                           |          | 
| 2   |                                 | 2) unable to update/save UMA resources                                                                      |   Open       |
| 3   |  USR   MGMT         | unable to add custom fields                                                                                           | Open          |
| 4   |                                | unable to patch SCIM user attribute                                                                                 | Open          |
| 5   |                  | in patching if one operation is valid and other is invalid then it is showing both operation invalid| Open          |
|  6   |JSON Web Key (JWK)|if we add invalid kty or kid and tried to access list of jwk , getting below error             | Open          |
| 7   |                  |            connection timeout (500 error ) after sometime( 1 day)                                                      |  Open          |
|  8  |                   |                  Jwt Signature is Invalid.                                                                                               |  Open

------------------------------------------------------------------------------------------------------------------------------------------------------------------------
[log.zip](https://github.com/JanssenProject/jans/files/8505108/log.zip)
[jans-cli-issues_1.xlsx](https://github.com/JanssenProject/jans/files/8505130/jans-cli-issues_1.xlsx)

**Comment by yuriyz** [comment]
@duttarnab we got conflict in `LicenseDetailsService`, see above. Please merge main into `[jans-jetty-upgrade](https://github.com/JanssenProject/jans/tree/jans-jetty-upgrade)` and fix `javax` -> `jakarta` if needed.

**Review by yuriyz — DISMISSED** [review]

**Review by moabu — APPROVED** [review]

**Comment by moabu** [comment]
Approved by @yurem @yuriyz and @moabu

**Review by yuriyz — APPROVED** [review]

**Review by moabu — APPROVED** [review]

**Comment by pujavs** [comment]
https://github.com/JanssenProject/jans/pull/1123#issuecomment-1101302279
1) UMA resource : This should be working now
2) USR MGMT: SCIM USR MGMT was removed and USR MGMT was added which should be working
3) JSON Web Key (JWK): This should be fixed now
@manojs1978 , Kindly test and confirm with the code from main branch

**Comment by manojs1978** [comment]
@pujavs above issues tested .
working fine now

**Pull request merged**